appinventor/components/src/com/google/appinventor/components/runtime/util/MediaUtil.java

```diff
@@ -202,6 +203,26 @@ public static boolean isExternalFile(Context context, String mediaPath) {
         || mediaPath.startsWith("/sdcard/") || isExternalFileUrl(context, mediaPath);
   }
 
+  /**
+   * Tests and asks file permission if the given path is a pathname pointing to
+   *                          an external file a
+   * @param path path of the media file
+   * @param handler permission handler that posts the results
+   * @return true if permission is needed
+   */
+
+  public static boolean askFilePermissionIfNeeded(Form form, String path, PermissionResultHandler handler) {
+    boolean needsFilePermission = RUtil.needsFilePermission(form, path, null);
+    Log.i(LOG_TAG, "RUtil, needs file permission = " + needsFilePermission);
+    if (needsFilePermission
+        && MediaUtil.isExternalFile(form, path)
```

Comment:
I'm not sure this particular check is needed given the logic of `RUtil.needsFilePermission`, since that function calls through to `FileUtil.isExternalStorageUri`.

Reply:
Oh yes, you are right we don't need to recheck it using `MediaUtil.isExternalFile`; which is also deprecated.